src/stepper/impl_rk4.f90

```diff
@@ -77,23 +84,30 @@ subroutine impl_rk4_step(self, t, dt)
             dt6 = f16 * dt
 
             ! set integrating factors
-            self%emq = dexp(- diss)
+            self%emq = dexp(- vdiss)
```

Comment:
ok, this looks like the integrating factor.  The other comments may pertain to vertical filtering?

Reply:
We have cn2.f90 (the integrator we used for the Beltrami paper) and impl_rk4.f90 (the integrator using integrating factors for diffusion).